services/messaging/src/main/java/org/eclipse/hono/messaging/SenderFactoryImpl.java

```diff
@@ -122,6 +122,14 @@ public void setConfiguration(final HonoMessagingConfigProperties config) {
                 LOG.debug("sender [{}] for container [{}] closed", address, connection.getRemoteContainer());
             }
         });
+        sender.detachHandler(remoteDetached -> {
+            if (remoteDetached.succeeded()) {
+                LOG.debug("sender [{}] detached (with closed=false) by peer [{}]", sender.getRemoteSource(), connection.getRemoteContainer());
+            } else {
+                LOG.debug("sender [{}] detached (with closed=false) by peer [{}]: {}", sender.getRemoteSource(), connection.getRemoteContainer(), remoteDetached.cause().getMessage());
+            }
+            sender.close();
```

Comment:
I do not think that this is enough, we also need to propagate the closing of the downstream link to the upstream protocol adapter. For that, we will probably need to be able to pass in a detach handler...